File: frontend/src/api/admin/maintenance.ts
import http from '@/api/http'

export type MaintenanceCounts = {
  visit_stat_days: number
  admin_logs: number
  captcha_challenges: number
  logs_dir: string
}

export type UploadPruneResult = {
  dry_run: boolean
  orphan_count: number
  total_bytes: number
  deleted_count: number
  freed_bytes: number
  sample_paths: string[]
}

export async function fetchMaintenanceCounts() {
  const { data } = await http.get<MaintenanceCounts>('/api/admin/maintenance/counts')
  return data
}

export async function cleanupVisitStats(olderThanDays: number) {
  const { data } = await http.post<{ visit_stats_deleted: number }>(
    '/api/admin/maintenance/cleanup/visit-stats',
    { older_than_days: olderThanDays },
  )
  return data
}

export async function cleanupAdminLogs(olderThanDays: number) {
  const { data } = await http.post<{ admin_logs_deleted: number }>(
    '/api/admin/maintenance/cleanup/admin-logs',
    { older_than_days: olderThanDays },
  )
  return data
}

export async function cleanupCaptcha() {
  const { data } = await http.post<{ captcha_deleted: number }>('/api/admin/maintenance/cleanup/captcha')
  return data
}

export async function scanUploadOrphans() {
  const { data } = await http.get<UploadPruneResult>('/api/admin/maintenance/upload-orphans')
  return data
}

export async function pruneUploadOrphans() {
  const { data } = await http.post<UploadPruneResult>('/api/admin/maintenance/prune-uploads', {
    execute: true,
    confirm_delete: true,
  })
  return data
}

File: backend/app/schemas/maintenance.py
from pydantic import BaseModel, Field


class MaintenanceCountsOut(BaseModel):
    visit_stat_days: int
    admin_logs: int
    captcha_challenges: int
    logs_dir: str = Field(description="滚动日志目录（相对 backend 或绝对路径）")


class VisitStatsCleanupIn(BaseModel):
    older_than_days: int = Field(..., ge=1, le=3650, description="删除早于此天数的按日 PV 汇总")


class AdminLogsCleanupIn(BaseModel):
    older_than_days: int = Field(..., ge=1, le=3650, description="删除早于此天数的管理操作日志")


class VisitStatsCleanupOut(BaseModel):
    visit_stats_deleted: int


class AdminLogsCleanupOut(BaseModel):
    admin_logs_deleted: int


class CaptchaCleanupOut(BaseModel):
    captcha_deleted: int


class UploadPruneBody(BaseModel):
    execute: bool = False
    confirm_delete: bool = False


class UploadPruneResultOut(BaseModel):
    dry_run: bool
    orphan_count: int
    total_bytes: int
    deleted_count: int
    freed_bytes: int
    sample_paths: list[str]
